nomad/job_endpoint.go

```diff
@@ -433,6 +438,53 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 	return nil
 }
 
+// specChanged checks to see if the job spec has changed. If the job is multiregion,
+// it checks all regions. This handles the case of a multiregion job that has been
+// stopped in one or more regions being restarted via the run command.
+func (j *Job) specChanged(existingJob *structs.Job, args *structs.JobRegisterRequest) (bool, error) {
+	if existingJob == nil {
+		return false, nil
+	}
+
+	if existingJob.SpecChanged(args.Job) {
+		return true, nil
+	}
+
+	if !existingJob.IsMultiregion() {
+		return false, nil
+	}
+
+	// If the spec hasn't changed, but this is a multiregion job, check other regions.
+	for _, region := range existingJob.Multiregion.Regions {
+		// Copy the job so we can mutate it by region and compare it with the response.
+		incomingJob := args.Job.Copy()
+		incomingJob.Region = region.Name
+		req := &structs.JobSpecificRequest{
+			JobID: incomingJob.ID,
+			QueryOptions: structs.QueryOptions{
+				Region:    region.Name,
+				Namespace: incomingJob.Namespace,
+				AuthToken: args.AuthToken,
+			},
+		}
+		resp := structs.SingleJobResponse{}
+		j.logger.Debug("checking spec for multiregion job")
+		err := j.GetJob(req, &resp)
+		if err != nil {
+			j.logger.Error("job lookup failed", "error", err)
+			return false, err
+		}
+
+		// If the jobspec changed or is nil (purged) for that region, assume it's being re-registered
+		// and treat like a spec change.
+		if resp.Job.SpecChanged(incomingJob) || resp.Job == nil {
```

Comment:
`SpecChanged` is safe if the receiver is nil but it's still better to flip this conditional so that the next person who reads this code doesn't have to _check_ that it is. 😁